src/collection.ts

```diff
@@ -1098,15 +1141,19 @@ export class Collection<TSchema extends Document = Document> {
    * @see https://docs.mongodb.com/manual/reference/operator/query/centerSphere/#op._S_centerSphere
    */
   countDocuments(): Promise<number>;
+  /** @deprecated Callbacks are deprecated and will be removed in the next major version */
   countDocuments(callback: Callback<number>): void;
   countDocuments(filter: Filter<TSchema>): Promise<number>;
+  /** @deprecated Callbacks are deprecated and will be removed in the next major version */
   countDocuments(callback: Callback<number>): void;
```

Comment:
can we remove this duplicate overload?

Reply:
It's not a type change if it's a duplicate overload.  it's literally a duplicate of line 1145

Reply:
An overload has is emitted to the definitions, removing it changes how the defintions resolve the correct call format. I agree we should remove it, I just want this PR to involve only docs changes, the plan is to generate the docs pages when we're done checking that the @deprecation tag is in all the right places

Reply:
This won't affect function call resolution at all - if that overload is matched, it will match the first instance of that overload.  If not, it will never match the second.

Can you explain the plan about generating the docs more?  Will that be included in this PR or in follow up work?

Reply:
In this PR I'll pull in the generated docs changes so we can merge it all together, but GH diff is difficult with many changes so I'm saving that for when the review is complete

Yea that's correct about the behavior at least to my understanding